test/e2e/lib/helpers/broker_event_transformation_test_with_source.go

```diff
@@ -235,65 +326,49 @@ func BrokerEventTransformationTestWithSchedulerSourceHelper(client *lib.Client,
 	}
 }
 
-func CreateKService(client *lib.Client) string {
+func CreateKService(client *lib.Client, imageName string) string {
 	client.T.Helper()
 	kserviceName := helpers.AppendRandomString("kservice")
 	// Create the Knative Service.
 	kservice := resources.ReceiverKService(
-		kserviceName, client.Namespace)
+		kserviceName, client.Namespace, imageName)
 	client.CreateUnstructuredObjOrFail(kservice)
 	return kserviceName
 
 }
 
-func createTriggerWithKServiceSubscriber(client *lib.Client, brokerName, kserviceName string) {
+func createTriggerWithKServiceSubscriber(client *lib.Client,
+	brokerName, kserviceName string,
+	triggerFilter eventingtestresources.TriggerOption) {
 	client.T.Helper()
 	// Please refer to the graph in the file to check what dummy trigger is used for.
-	dummyTriggerName := "dummy-broker-" + brokerName
+	triggerName := "trigger-broker-" + brokerName
 	client.Core.CreateTriggerOrFail(
-		dummyTriggerName,
+		triggerName,
 		eventingtestresources.WithBroker(brokerName),
-		eventingtestresources.WithAttributesTriggerFilter(
-			eventingv1alpha1.TriggerAnyFilter, eventingv1alpha1.TriggerAnyFilter,
-			map[string]interface{}{"type": "e2e-testing-dummy"}),
+		triggerFilter,
 		eventingtestresources.WithSubscriberServiceRefForTrigger(kserviceName),
 	)
 }
 
-func createTriggerWithTargetServiceSubscriber(client *lib.Client, brokerName, targetName string) {
+func createTriggerWithTargetServiceSubscriber(client *lib.Client,
+	brokerName, targetName string,
+	triggerFilter eventingtestresources.TriggerOption) {
 	client.T.Helper()
 	respTriggerName := "resp-broker-" + brokerName
 	client.Core.CreateTriggerOrFail(
 		respTriggerName,
 		eventingtestresources.WithBroker(brokerName),
-		eventingtestresources.WithAttributesTriggerFilter(
-			eventingv1alpha1.TriggerAnyFilter, eventingv1alpha1.TriggerAnyFilter,
-			map[string]interface{}{"type": "e2e-testing-resp"}),
+		triggerFilter,
 		eventingtestresources.WithSubscriberServiceRefForTrigger(targetName),
 	)
 }
 
-func createTriggersAndKService(client *lib.Client, brokerName, targetName string) {
-	client.T.Helper()
-	// Create the Knative Service.
-	kserviceName := CreateKService(client)
-
-	// Create a Trigger with the Knative Service subscriber.
-	createTriggerWithKServiceSubscriber(client, brokerName, kserviceName)
-
-	// Create a Trigger with the target Service subscriber.
-	createTriggerWithTargetServiceSubscriber(client, brokerName, targetName)
-
-	// Wait for ksvc, trigger ready.
-	client.Core.WaitForResourceReadyOrFail(kserviceName, lib.KsvcTypeMeta)
-	client.Core.WaitForResourcesReadyOrFail(eventingtestlib.TriggerTypeMeta)
-}
-
 func makeTargetJobOrDie(client *lib.Client, targetName string) {
 	client.T.Helper()
 	job := resources.TargetJob(targetName, []v1.EnvVar{{
-		Name:  "TARGET",
-		Value: "falldown",
+		Name:  "TIME",
```

Comment:
can we comment what this time is for?

Reply:
`Time` will get deleted after I change the `target` to the test framework that other targets use like `storage_target`